network/p2p/p2p.go

```diff
@@ -95,8 +95,8 @@ type p2pNetwork struct {
 
 	backoffConnector *libp2pdiscbackoff.BackoffConnector
 
-	fixedSubnets  []byte
-	activeSubnets []byte
+	fixedSubnets  records.Subnets
+	activeSubnets records.Subnets
```

Comment:
Maybe add comments clarifying what these are (in particular `fixedSubnets`) while we are at it.